File: Docs/Card_game_ref/BlackJack/lib/deck.rb
require_relative 'card'

# Represents a deck of playing cards.
class Deck
  # Returns an array of all 52 playing cards.
  def self.all_cards
    cards = []
     Card.suits.each do |suit|
      Card.values.each do |value|
        cards << Card.new(suit,value)
      end
    end
    cards
  end



  def initialize(cards = Deck.all_cards)
    @deck = cards
  end

  # Returns the number of cards in the deck.
  def count
    @deck.length
  end

  # Takes `n` cards from the top of the deck.
  def take(n)
    raise "not enough cards" if n > @deck.length
    taken = []
     n.times do
       taken << @deck.shift
     end
     taken
  end

  # Returns an array of cards to the bottom of the deck.
  def return(cards)
    cards.each do |card|
    @deck << card
    end
  end
end

File: js/deck.js
const Card = require("./card");

class Deck {
  constructor(){
    this.values = ['2','3','4','5','6','7','8','9','10','J','Q','K','A'];
    this.suits = ['H','D','S','C'];
    this.cards = [];

    for (var s = 0; s < this.suits.length; s++) {
      for (var v = 0; v < this.values.length; v++) {
        this.cards.push( new Card( this.suits[s], this.values[v] ))
      }
    }
    this.cards = this.shuffle(this.cards);
 };

 shuffle(a) {
  for (let i = a.length; i; i--) {
      let j = Math.floor(Math.random() * i);
      [a[i - 1], a[j]] = [a[j], a[i - 1]];
  }
  return a;
}

 take(n){
  let taken = [];
  for (var i = 0; i < n; i++) {
    taken.push(this.cards.shift());
  }
  this.outOfCards();
  return taken;
  //returns cards in a array
 }

 outOfCards(){
   if (this.cards.length === 0){
      let shuffleUp = new Deck
      this.cards = shuffleUp.cards
   }
 }

}

module.exports = Deck;
